src/hypergol/base_tensorflow_model_block.py

```diff
@@ -0,0 +1,37 @@
+# pylint: disable=E0611, W0235
+
+import inspect
+import json
+from tensorflow.python.keras import layers
+
+
+class BaseTensorflowModelBlock(layers.Layer):
+    """Subclasses tensorflow-keras layers to provide logical groupings of functionality/layers."""
+
+    def __init__(self, **kwargs):
+        super(BaseTensorflowModelBlock, self).__init__(**kwargs)
+
+    def get_config(self):
+        parameters = {k: v for k, v in self.__dict__.items() if k in inspect.signature(self.__class__).parameters.keys()}
+        config = super(BaseTensorflowModelBlock, self).get_config()
+        for name, value in parameters.items():
+            config.update({name: value})
+        return config
+
+    def get_name(self):
+        return self.__class__.__name__
+
+    def save_to_dictionary(self, directory):
+        json.dump(self.get_config(), open(f'{directory}/{self.get_name()}.json', 'w'))
```

Comment:
these should be in model manager so rewriting the model manager enables to store the model somewhere else.

Reply:
Good point!